proxy_agent/src/redirector/linux.rs

```diff
@@ -505,6 +507,99 @@ fn lookup_audit_internal(bpf: &Bpf, source_port: u16) -> std::io::Result<AuditEn
     }
 }
 
+pub fn update_wire_server_redirect_policy(redirect: bool) {
+    update_redirect_policy_internal(
+        constants::WIRE_SERVER_IP_NETWORK_BYTE_ORDER,
+        constants::WIRE_SERVER_PORT,
+        redirect,
+    );
+}
+
+pub fn update_imds_redirect_policy(redirect: bool) {
+    update_redirect_policy_internal(
+        constants::IMDS_IP_NETWORK_BYTE_ORDER,
+        constants::IMDS_PORT,
+        redirect,
+    );
+}
+
+fn update_redirect_policy_internal(dest_ipv4: u32, dest_port: u16, redirect: bool) {
+    unsafe {
```

Comment:
I would recommend not wrapping this whole block in `unsafe`.

The convention in Rust projects with unsafe is to use on the smallest, most minimal block possible, and to include a documentation block on why it's safe to do. See, for example, the [usage in the standard library](https://github.com/rust-lang/rust/blob/e8fbd991287f637f95016a71ddc13438415bbe59/library/core/src/fmt/float.rs#L48) - the reason the code within the `unsafe` block is okay should be documented (assumptions made, potential ways those assumptions could be broken, etc).

It's extremely easy to introduce safety issues when inside an `unsafe` block as you're promising the compiler you'll manually check a lot of the things it automatically checks for you, so making them small and isolated is best.